examples/generative/finetune_stable_diffusion.py

```diff
@@ -0,0 +1,507 @@
+Accelerator: GPU
+"""
+"""
+## Introduction
+
+This tutorial shows how to fine-tune a
+[Stable Diffusion model](https://keras.io/guides/keras_cv/generate_images_with_stable_diffusion/)
+on a custom dataset containing {image,caption} pairs. We build on top of the fine-tuning
+script provided by Hugging Face
+[here](https://github.com/huggingface/diffusers/blob/main/examples/text_to_image/train_text_to_image.py).
+
+We assume that you have a high-level understanding of the Stable Diffusion model.
+Following resources can be helpful if you're looking for resources in that regard:
+
+* [High-performance image generation using Stable Diffusion in KerasCV](https://keras.io/guides/keras_cv/generate_images_with_stable_diffusion/)
+* [Stable Diffusion with Diffusers](https://huggingface.co/blog/stable_diffusion)
+
+It's highly recommended that you use a GPU having at least 30 GBs of memory to execute
+the code.
+
+By the end of the guide, you'll be able to generate images of interesting pokemons:
+
+![custom-pokemons](https://i.imgur.com/X4m614M.png)
+
+For the code, the tutorial relies on KerasCV 0.3.5 which is not yet available on PyPI.
+So, we need to install it from the source. Additionally, we need the latest stable 
+version of TensorFlow.
+"""
+
+"""shell
+pip install git+https://github.com/keras-team/keras-cv -q
+pip install -U tensorflow -q
+"""
+
+"""
+## What are we fine-tuning? 
+
+A Stable Diffusion model can be decoupled into several key models:
+
+* A text encoder that projects the input prompt to a latent space.
+* A variational autoencoder (VAE) that projects an input image into a latent space acting
+as an image encoder.
+* A diffusion model that refines an input latent and produces another latent in the
+presence of the encoded text prompt.
+* A decoder that generates images given a latent from the diffusion model.
+
+It's worth noting that during the process of generating an image from a text prompt, the
+image encoder is not typically employed.
+
+However, during the process of fine-tuning, the workflow goes like the following:
+
+1. An input prompt is projected to a latent space by the text encoder. The caption
+associated with an image is referred to as the prompt.
+2. An input image is projected to a latent space by the image encoder portion of the VAE.
+3. A small amount of noise is added to the image latent with a given timestep.
+4. The diffusion model uses latents from these two spaces along with a timestep embedding
+to predict the noise that was added to the image latent.
+5. A reconstruction loss is calculated between the predicted noise and the original noise
+added in step 3.
+6. Finally, the diffusion model parameters are then optimized w.r.t this loss using
+gradient descent.
+
+Note that only the diffusion model parameters are updated during fine-tuning while the
+text and the image encoders are kept frozen.
+
+Don't worry if this reads complicated. The code is much simpler than this.
```

Comment:
not related to this PR, but we should maybe consider providing a train-from-scratch example under KerasCV